### PR TITLE
Merge remote-tracking branch 'origin/refactor_key_builder.rs' into refactor_key_builder.rs

### PR DESCRIPTION
**Pull request by Damienkatz**
Stop using “keypath” in function signatures and instead use “kp”. Use
constants for key prefixes instead of literals, giving one place to
look to what a key means. More consistent names for functions.

### CODE REVIEW

File: src/key_builder.rs [hunk 1]
```diff
@@ -328,49 +354,57 @@ impl KeyBuilder {
         self.keypath.push(escaped_key);
     }
 
+    /// adds array to keypath
```
Comment (vmx) [inline src/key_builder.rs]: Minor nitpick. I always start comments uppercase and most are. If you don't feel like it, you don't have to change it, it's just an info for future comments.

Comment (Damienkatz) [inline src/key_builder.rs]: That rename is outside the scope of this patch.
